### ISSUE TITLE
Duplicates comment added on top of a `part` declaration

### ISSUE THREAD

**Issue by renatoathaydes** [dart-lang/dart_style#812]
I have a dart file like this:

```dart
// main.dart
import 'package:yamlgen/yamlgen.dart';

/// declare that the generated file, <this_file.g.dart>, is a part of this file.
part 'main.g.dart';
```

When I format the code, dartfmt duplicates the comment and I end up with this:

```dart
// main.dart
import 'package:yamlgen/yamlgen.dart';

/// declare that the generated file, <this_file.g.dart>, is a part of this file.
part 'main.g.dart';

/// declare that the generated file, <this_file.g.dart>, is a part of this file.
```

Doing format again does it again:

```dart
// main.dart
import 'package:yamlgen/yamlgen.dart';

/// declare that the generated file, <this_file.g.dart>, is a part of this file.
part 'main.g.dart';

/// declare that the generated file, <this_file.g.dart>, is a part of this file.

/// declare that the generated file, <this_file.g.dart>, is a part of this file.
```

**Comment by renatoathaydes** [comment]
Related:

I was using a normal comment before with only 2 slashes... but then the formatter puts it in the wrong place:

This:

```dart
// main.dart
import 'package:yamlgen/yamlgen.dart';

// declare that the generated file, <this_file.g.dart>, is a part of this file.
part 'main.g.dart';
```

After formatting becomes this:

```dart
// main.dart
import 'package:yamlgen/yamlgen.dart';

part 'main.g.dart';
// declare that the generated file, <this_file.g.dart>, is a part of this file.

```

Quite annoying, why does it need to change the comment position?

**Comment by munificent** [comment]
I can't repro this and if dartfmt *did* do this, it would trip some of its own internal failsafes that ensure the output only differs from the input in whitespace, so I don't think this is coming from dartfmt.

Are you invoking it through some editor or IDE integration? If so, something in there is probably causing the problem. Take a look at #767 and see if any of that sounds familiar.

**Comment by renatoathaydes** [comment]
Hi @munificent sorry that I blamed dartfmt for this :) I was under the assumption that IntelliJ formats code by calling `dartfmt` but it does not seem that's the case?

I was using the IntelliJ Dart Plugin version 191.7019. Upgrading to 191.7221 didn't solve the problem.

I did some research, and it seems that by default, IntelliJ will delegate to `dartfmt` but due to [this bug](https://youtrack.jetbrains.com/issue/WEB-35206) it looks like that option was not working before for me.

The IntelliJ's formatter seems to be still broken. I filed a bug with Jetbrains: https://youtrack.jetbrains.com/issue/IDEA-214757

Thanks for checking.
